src/mpp/Dec.hpp

```diff
@@ -1172,6 +1172,21 @@ bool jump_read_key(BUF& buf, T... t)
 	return jump_find_key<PAIRS, FAMILY, PATH>(val, IS, buf, t...);
 }
 
+template<compact::Family FAMILY, size_t SUBRULE,
+	 class DST, class BUF, class = void>
+struct is_object_readable_by_value : std::false_type {
+};
+
+template<compact::Family FAMILY, size_t SUBRULE,
+	 class DST, class BUF>
+struct is_object_readable_by_value<FAMILY, SUBRULE, DST, BUF,
+				   typename std::enable_if_t<rule_by_family_t<FAMILY>::is_readable_by_value && std::is_assignable_v<DST, decltype(read_value<FAMILY, SUBRULE>(std::declval<BUF>()))>, bool>::type> : std::true_type {
+};
+
+template<compact::Family FAMILY, size_t SUBRULE,
+	 class DST, class BUF>
+constexpr bool is_object_readable_by_value_v = is_object_readable_by_value<FAMILY, SUBRULE, DST, BUF>::value;
```

Comment:
This trait is combination of two booleans. There is no need in the helper and `std::enable_if`:
```c++
template<compact::Family FAMILY, size_t SUBRULE, class DST, class BUF>
constexpr bool is_object_readable_by_value_v = rule_by_family_t<FAMILY>::is_readable_by_value &&
        std::is_assignable_v<DST, decltype(read_value<FAMILY, SUBRULE>(std::declval<BUF>()))>;
```

Reply:
Fixed, thanks!